tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1293,7 +1293,8 @@ public:
       }
       TDirectory *treeDirectory = fOutputFiles[slot].get();
       if (!fDirName.empty()) {
-         treeDirectory = fOutputFiles[slot]->mkdir(fDirName.c_str());
+         // call exist_ok=true since MT can end up making this call multiple times
+         treeDirectory = fOutputFiles[slot]->mkdir(fDirName.c_str(), "", true);
```

Comment:
@bluehood `mkdir` appears in `InitTask`. Print statements reveal that `InitTask` gets called multiple times for a given slot. Moreover, the number of times it gets called varies; I can call `Snapshot` several times in a row, and see that `InitTask` was called for slow 0 one, two, or three times.

If I have it print the number of entries when `fOutputFiles[slot]->Write` is called for slot 0, I can see that `Write` is called every time `InitTask` is called and that `fOutputFiles[0]` always has 1250 entries.

I hope this helps. I'm not sure where else to look.

Reply:
Hi @goi42 , this is certainly useful! `InitTask` can indeed be called multiple times per thread.

Looking at the code and your changes, I think this can go in safely.